lib/beaker/dsl/structure.rb

```diff
@@ -175,10 +179,17 @@ def confine(type, criteria, host_array = nil, &block)
         hosts_to_modify = host_array || hosts
         case type
         when :except
-          hosts_to_modify = hosts_to_modify - select_hosts(criteria, hosts_to_modify, &block)
+          if criteria and ( not criteria.empty? )
+            hosts_to_modify = hosts_to_modify - select_hosts(criteria, hosts_to_modify, &block)
+          else
+            # confining to all hosts *except* provided array of hosts
+            hosts_to_modify = hosts - host_array
+          end
         when :to
           if criteria and ( not criteria.empty? )
             hosts_to_modify = select_hosts(criteria, hosts_to_modify, &block)
+          else
+            # confining to only hosts in provided array of hosts
```

Comment:
Yup, just wanted the comment in there to be super clear what the fall through case is.  Not a coding syntax that you like?